src/Storages/MergeTree/MergeTreeMutationEntry.cpp

```diff
@@ -72,6 +72,12 @@ MergeTreeMutationEntry::MergeTreeMutationEntry(MutationCommands commands_, DiskP
             TransactionID::write(tid, *out);
             *out << "\n";
         }
+        if (commands.settings_ast)
+        {
+            *out << "settings: ";
+            commands.writeSettings(*out);
+            *out << "\n";
+        }
```

Comment:
Settings are written unconditionally, so you will not be able to rollback to older version once you run a mutation. We must mention it in the changelog and maybe we should change the category to "backward incompatible change", so users will pay attention.

Reply:
Fixed (settings are only written if needed).

Reply:
Now, settings are written only when used, so it's less of a backwards compatibility issue. I also updated the PR description (hoping it goes into the changelog) to reflect this. Do you still think I should change to "Backward incompatilble change"?

Reply:
No, "Improvement" (or "New Feature") is more suitable now. Changes do not lead to any compatibility issues until you set the setting to 1. And it's well-known fact that it may not be possible to rollback to older versions since you explicitly started to use new features/settings.